src/dreyfus_rpc.erl

```diff
@@ -47,29 +47,31 @@ call(Fun, DbName, DDoc, IndexName, QueryArgs0) ->
     {_LastSeq, MinSeq} = calculate_seqs(Db, Stale),
     case dreyfus_index:design_doc_to_index(DDoc, IndexName) of
         {ok, Index} ->
-            case dreyfus_index_manager:get_index(DbName, Index) of
-                {ok, Pid} ->
-                    case dreyfus_index:await(Pid, MinSeq) of
-                        {ok, IndexPid, _Seq} ->
-                            Result = dreyfus_index:Fun(IndexPid, QueryArgs),
-                            rexi:reply(Result);
-                        % obsolete clauses, remove after upgrade
-                        ok ->
-                            Result = dreyfus_index:Fun(Pid, QueryArgs),
-                            rexi:reply(Result);
-                        {ok, _Seq} ->
-                            Result = dreyfus_index:Fun(Pid, QueryArgs),
-                            rexi:reply(Result);
-                        Error ->
-                            rexi:reply(Error)
-                    end;
+            {ok, Pid} = get_index_pid(DbName, Index),
+            case dreyfus_index:await(Pid, MinSeq) of
+                {ok, IndexPid, _Seq} ->
+                    Result = dreyfus_index:Fun(IndexPid, QueryArgs),
+                    rexi:reply(Result);
+                % obsolete clauses, remove after upgrade
+                ok ->
+                    Result = dreyfus_index:Fun(Pid, QueryArgs),
+                    rexi:reply(Result);
+                {ok, _Seq} ->
+                    Result = dreyfus_index:Fun(Pid, QueryArgs),
+                    rexi:reply(Result);
                 Error ->
                     rexi:reply(Error)
             end;
         Error ->
             rexi:reply(Error)
     end.
 
+get_index_pid(DbName, Index) ->
```

Comment:
Please remove the extraction of get_index_pid from this PR so we can focus on the purge feature change.